### ISSUE TITLE
Can make a type optional multiple times, as in `token: str????;`

### ISSUE THREAD

**Issue by skyrpex** [winglang/wing#4678]
### I tried this:

```ts
struct ClientProps {
  token: str????;
  orgSlug: str?;
}
```

### This happened:

The struct compiles correctly but later usage fails, for example:

```ts
struct ClientProps {
  token: str???;
  orgSlug: str?;
}

class Client {
  token: str;

  init(props: ClientProps?) {
    // Interestingly, the `if let` removes one of the parentheses but leaves the rest :)
    if let token = props?.token {
      this.token = token;
      //           ^ Expected type to be "str", but got "str??" instead
preflight token: str??
    }
  }
}
```

### I expected this:

The original snippet should fail to compile, I guess.

### Component

Compiler

### Wing Version

0.41.0

### Community Notes

* Please vote by adding a 👍 reaction to the issue to help us prioritize.
* If you are interested to work on this issue, please leave a comment.

**Comment by staycoolcall911** [comment]
I agree, let's forbid it. I don't see a reason to allow multiple optionality modifiers.
CC @eladb , @yoav-steinberg, @MarkMcCulloh, @Chriscbr, @hasanaburayyan

**Comment by yoav-steinberg** [comment]
Logically I see no problem with the current behavior. Disallowing nested optionality _might_ be a good idea just to reduce complexity and bugs, on the other hand there might be some edge cases which will break because of this, for example if we add type aliases imagine this:
```wing
alias maybe_string = str?;
let x: maybe_string? = nil;
// now x is a nested optional
```
Waiting for other inputs to form a concrete opinion here...

**Comment by ekeren** [comment]
@yoav-steinberg 

What is the use case for an optional `maybe_string`, can't we output a compilation error in this case
```
let x: maybe_string? = nil;
//                             ^ can't nest optional, `maybe_string` is already an optional type
```

**Comment by yoav-steinberg** [comment]
> What is the use case for an optional `maybe_string`, can't we output a compilation error in this case

I can't think of a good one, but these things happen. If we're sure there's no good use case we can remove support for this. But I'd like more people to be sure of this, the current logic is sound and is supported by other langs, so we should be careful changing it. This is valid swift code:
```swift
let x: Int?? = Optional.some(nil)
```

**Comment by MarkMcCulloh** [comment]
Wing's current optional works like typescript, which IMO works for us:

`str?` is basically `str | nil`

This means that:

`str??` is `str | nil | nil`, which is just `str | nil`

So there's not much use in allowing multiple `?`.
I guess there are reasons that you may want nested options in your API, but IMO they are uncommon and maybe not worth encouraging.

**Comment by Chriscbr** [comment]
I think for now we can disallow multiple `?` this since the syntax doesn't have any extra meaning.
